templates/layout.mustache

```diff
@@ -18,11 +18,31 @@
 <head>
 	<meta charset="utf-8">
   <title>{{getTitle}}</title>
+	<meta property="og:title" content="{{getTitle}}">
+	<meta property="og:description" content="{{getLinkTitle .}}">
+  <meta property="og:image" content="https://www.bitovi.com/hubfs/open-source/os-donejs.jpg">
 	<meta name="google-site-verification" content="-E1_on_BhUFqVXxNgtGgNYF5FIJojlOksLPK8zdeiL8" />
-	<meta name="description" content="DoneJS is an open source JavaScript framework that makes it easy to build high performance, real time web and mobile applications.">
+	<meta name="description" content="{{getLinkTitle .}}">
 	<meta name="author" content="Bitovi - DoneJS">
 	<meta http-equiv="X-UA-Compatible" content="IE=edge">
 	<meta name="viewport" content="width=device-width, initial-scale=1">
+  <script type="application/ld+json">
+		{
+			"@context": "http://www.schema.org",
+			"@type": "SoftwareSourceCode",
+			"applicationCategory": "DeveloperApplication",
+			"brand": "Bitovi",
+			"category": "JavaScript Frameworks",
+			"codeRepository": "https://github.com/donejs/donejs",
+			"description": "{{getLinkTitle .}}",
+			"image": "https://www.bitovi.com/hubfs/open-source/os-donejs.jpg",
+			"license": "https://github.com/donejs/donejs/blob/master/license.md",
+			"logo": "https://www.bitovi.com/hubfs/open-source/os-donejs.jpg",
+			"name": "{{getDocumentTitle .}}",
+			"programmingLanguage": "JavaScript",
+			"softwareVersion" : "3.2.0"
```

Comment:
Was what we did for CanJS not doable here? https://github.com/canjs/bit-docs-html-canjs/blob/54d3a51250d56fe1cf2c65309e487abc8a58604c/templates/layout.mustache#L26

If not, I’d prefer to remove this field rather than have the version be out of date.

Reply:
The data structure is not the same as in canjs, maybe this can be fixed in the docs.

Reply:
Ok, so there’s no package/version info available?

Reply:
For the theme we have the version but not for the `donejs` itself

Reply:
It’s not in the docMap? What does the entry in the docMap look like?

Reply:
```
{ content: 'long content',
  src: { path: 'docs/guides/setting-up.md' },
  body: '\n## Windows\n\n### Prerequisites\n\nThis will help you get set up with DoneJS on Windows. To use DoneJS you need a C++ compiler (for native dependencies). First you need a recent version of [Node.js](https://nodejs.org/en/).\n\n#### Node.js\n\nTo download Node.js visit [https://nodejs.org](https://nodejs.org). There will be two big green buttons for the two recommended downloads. Choose the one on the right, which says **Current**. This will give you the most recent version of Node.js.\n\nOnce downloaded, double-click the installer and go through the installation process. Once complete you should be able to open up a terminal and type:\n\n```shell\nnode -v\n```\n\nWhich should then print out the version of Node.js that is installed.\n\n#### Package Management\n\nIn this guide we\'ll use [chocolatey](https://chocolatey.org/) to install packages needed. You don\'t have to use chocolatey if you don\'t want, and can instead search for the dependencies and install them with a Windows installer, but we\'ll use chocolately because it makes things a bit easier.\n\nAfter you\'ve installed chocolatey by following the instructions [on the homepage](https://chocolatey.org/) **open an administrative console** and proceed to the next step.\n\n#### Python 2.x\n\nNative dependencies in Node.js are installed with [node-gyp](https://github.com/nodejs/node-gyp) which uses Python as a build tool. It expects Python 2.x:\n\n```shell\nchoco install python2 -y\n```\n\n#### Windows SDK\n\nNext we need the Windows SDK. We\'re going to assume Windows 7, but adjust this command to the version of Windows you use (for Windows 10 it is windows-sdk-10.0):\n\n```shell\nchoco install windows-sdk-7.1 -y\n```\n\n#### Visual Studio Express\n\nInstalling Visual Studio Express gives us the C++ compiler we need. If you are using Windows 10 or get an error with this command you can also download Visual Studio Express [here](https://www.visualstudio.com/en-us/products/visual-studio-express-vs.aspx):\n\n```shell\nchoco install visualstudioexpress2013windowsdesktop -y\n```\n\n#### OpenSSL\n\nIf you plan on using Continuous Deployment with Travis CI, the [OpenSSL](https://slproweb.com/products/Win32OpenSSL.html) software is needed to encrypt your CDN API key.\n\n### Environmental Variables\n\nIn order to switch to production mode you need to set the environmental variable `NODE_ENV`. Depending on which console you use this can be done in one of two ways:\n\n**Command Prompt**\n\n```\nset NODE_ENV=production\n```\n\n**Powershell**\n\n```\n$env:NODE_ENV="production"\n```\n\nTo later remove these environment variables:\n\n**Command Prompt**\n\n```\nset NODE_ENV=\n```\n\n**Powershell**\n\n```\n$env:NODE_ENV=""\n```\n\n\n### Android Development\n\nIn order to develop an Android application you need to install the [Android Studio](https://developer.android.com/sdk/index.html). The installer will prompt you to also install Java if you don\'t already have it.\n\n<img src="static/img/wiz-1.png" alt="Android Studio installation" style="width: 100%;" class="alignnone size-full wp-image-2665" />\n\nWhen you start Android Studio for the first time it will go through the installation of the SDK and related tools. If prompted select to install:\n\n* Android 7.x\n* Android SDK\n* Performance (this will improve the emulator start time)\n* Android Virtual Device (this will create an optimized emulator)\n\nOnce this is complete you should have everything you need for Android development. You can close Android Studio as you won\'t need it for DoneJS development.\n\n<a id="mac-os-x"></a>\n## macOS\n\n### Node.js\n\nTo download Node.js visit [https://nodejs.org](https://nodejs.org). There will be two big green buttons for the two recommended downloads. Choose the one on the right, which says **Current**. This will give you the most recent version of Node.js.\n\nOnce downloaded, double-click the installer and go through the installation process. Once complete you should be able to open up a terminal and type:\n\n```shell\nnode -v\n```\n\nWhich should then print out the version of Node.js that is installed.\n\n### Prerequisites\n\nTo get DoneJS working on macOS you need Xcode command line tools which you can get by typing:\n\n```shell\nxcode-select --install\n```\n\nAnd to build iOS apps, after installing Node, install the `ios-sim` package with:\n\n```\nnpm install -g ios-sim\n```\n\n## Debian / Ubuntu\n\nInstalling in a Debian / Ubuntu environment takes a little extra work because the version of Node shipped is older than what is supported by DoneJS (and most other Node-based software).\n\n### Prerequisites\n\nInstead of installing Node.js from the repository we recommend using a PPA (a repository maintained by a 3rd party). First get a copy of `curl` if you don\'t already have it:\n\n```\nsudo apt-get install curl\n```\n\nThen add the PPA to your source list:\n\n```\ncurl -sL https://deb.nodesource.com/setup_0.12 | sudo bash -\n```\n\nInstall Node.js:\n\n```\nsudo apt-get install nodejs\n```\n\nIt\'s important to also install the `build-essential` package afterwards. This will provide you the C++ compiler needed to build the native dependencies:\n\n```\nsudo apt-get install build-essential\n```\n\n### Android Development\n\nIn order to develop an Android application you need to install the [Android Studio](https://developer.android.com/sdk/index.html). Once download untar/gzip it:\n\n```\ntar xvf android-sdk.X-linux.tgz\n```\n\nThis will create an `android-sdk-linux` folder. At this point you might want to move it somewhere else, `$HOME/lib/android-sdk-linux` is a good place.\n\nAdd the android-sdk-linux/tools folder to your `PATH` with: `export PATH="path/to/android-sdk-linux/tools`. Add this to your `~/.bashrc` or `~/.zshrc` config so it will persist.\n\nAdditionally the `ANDROID_HOME` environmental variable needs to be set. Set it with:\n\n```\nexport ANDROID_HOME="/home/name/lib/android-sdk-linux"\n```\n\nThis too should be added to your `~/.bashrc` or `~/.zshrc`.\n\n#### Java\n\nIf you don\'t already have a Java JDK installed you can do so with:\n\n```\nsudo apt-get install default-jdk\n```\n\n#### Platform and Build Tools\n\nOnce you\'ve installed Android Studio you still have a few things to do. You need to install the Android SDK Platform and Build tools. From the command-line run:\n\n```\n$ANDROID_HOME/tools/android\n```\n\nThis starts the Android SDK Manager. From this screen you can select:\n\n* Android 6.0\n* Android SDK Build-tools (23+)\n* Intel x86 Emulator Accelerator (this will improve the emulator start time)\n\nClick all of these and anything else you need and click Install packages.\n\n#### Virtual Device Manager\n\nFrom the command-line run:\n\n```\n$ANDROID_HOME/tools/android avd\n```\n\nThis starts the Android Virtual Device (AVD) Manager. This is used to manager virtual devices that will run in the emulator.\n\nClick **Create** and make sure to fill out:\n\n* AVD Name (this can be whatever you want)\n* Device\n* Target (the API level you installed)\n* CPU (try an Intel CPU if possible)\n\nThen click **OK** to create the device.\n\nClose the AVD Manager and you should have everything you need for Android development.\n\n\n## Vagrant & VirtualBox\n\n### Prerequisites\n\nFirst at all, download and install [**VirtualBox**](https://www.virtualbox.org/) and [**Vagrant**](https://www.vagrantup.com/). Once VirtualBox and Vagrant have been installed, you should add the DoneJS box to your Vagrant installation.\n\n### Installing the DoneJS Vagrant Box\nOpen your command prompt and type\n```\nvagrant --version\n```\nto see if Vagrant is available in your terminal.\n\nWithin your terminal change to the folder you would like to install the Vagrant Box. Using the following command for adding the box:\n```\nvagrant box add Juke/DoneJS\n```\n\nVagrant will ask you for which provider you will adding the box. Choose: **Virtualbox**\nVagrant will now downloading the latest version of the DoneJS development environment. It will take a few minutes, depending on your Internet connection speed.\n\nOnce Vagrant has successfully finished downloading all the file, you can now initialize the DoneJS Box by typing\n```\nvagrant init Juke/DoneJS\n```\n\nVagrant is creating a **Vagrantfile** in your folder. Your custom configuration for the machine can be done in this file.\nFor more information check out https://www.vagrantup.com/docs/vagrantfile/\n\n\n### Configure your shared folders\nBefore you start the Vagrant Box, you have to specify which folder you would like to sync from your local machine to the virtual machine.\nFor that, open the **Vagrantfile** in that folder you initialized Vagrant. Scroll down to:\n```\n# Share an additional folder to the guest VM. The first argument is\n# the path on the host to the actual folder. The second argument is\n# the path on the guest to mount the folder. And the optional third\n# argument is a set of non-required options.\n# config.vm.synced_folder "../data", "/vagrant_data"\n```\nadd a new Synced Folder configuration below that may look like this\n```\nconfig.vm.synced_folder "C:/www/donejs", "/home/vagrant/donejs"\n```\n\nMake sure that `C:/www/donejs` is a valid directory.\n\nFor more information check the [documentation](https://www.vagrantup.com/docs/synced-folders/basic_usage.html).\n\n### Launching the Vagrant Box\nOnce you have done all the configuration run\n```\nvagrant up\n```\ncommand from your DoneJS Vagrant Box directory\n\n### Connecting via SSH\nTo connect to your DoneJS environment via SSH, enter the\n```\nvagrant ssh\n```\ncommand in your terminal.\n\n### Further information\nFor further information about what is pre-installed on the Vagrant Box and which ports are forwarding to your host please check out the Github repository [DoneJS Vagrant](https://github.com/donejs/donejs-vagrant)\n\n',
  description: 'This page contains information on setting up DoneJS. It will walk you through getting prerequisites needed before you ever install DoneJS and includes platform-specific pieces of information not covered in the Quick Start or In Depth guides.\n\nDoneJS officially supports:\n\n - [Node](https://nodejs.org) 6 and up\n - [npm](https://www.npmjs.com/) 3.x and up\n\n',
  name: 'SettingUp',
  title: 'Setting Up DoneJS',
  type: 'page',
  parent: 'Guides',
  hideSidebar: true,
  outline: { depth: 2, tag: 'ol' },
  comment: ' ' }
```

Reply:
That looks like the entry for the current page. From there, something like this should be possible, to walk up from that object to the root, or whatever has a `package` property on it. https://github.com/canjs/bit-docs-html-canjs/blob/54d3a51250d56fe1cf2c65309e487abc8a58604c/templates/helpers.js#L177-L203

Maybe the root entry doesn’t have `package`… but I would expect it does if it’s being generated like the docMap for CanJS.